solutions/USACO open16-splitting-the-field.mdx

```diff
@@ -92,238 +92,65 @@ From the analysis:
-		}
-		res += "}"; return res;
-	#else
-		bool fst = 1; str res = "";
-		for (const auto& x: v) {
-			if (!fst) res += " ";
-			fst = 0; res += ts(x);
-		}
-		return res;
+//pairs
+using pi = pair<int,int>;
+#define f first
+#define s second
+#define mp make_pair
 
-	#endif
-}
-tcTU> str ts(pair<T,U> p) {
-	#ifdef LOCAL
-		return "("+ts(p.f)+", "+ts(p.s)+")"; 
-	#else
-		return ts(p.f)+" "+ts(p.s);
-	#endif
+void setIO(string name = "") {
+	cin.tie(0)->sync_with_stdio(0); // see /general/fast-io
+	if (sz(name)) {
+		freopen((name+".in").c_str(), "r", stdin); // see /general/io
+		freopen((name+".out").c_str(), "w", stdout);
+	}
 }
 
-// OUTPUT
-tcT> void pr(T x) { cout << ts(x); }
-tcTUU> void pr(const T& t, const U&... u) { 
-	pr(t); pr(u...); }
-void ps() { pr("\n"); } // print w/ spaces
-tcTUU> void ps(const T& t, const U&... u) { 
-	pr(t); if (sizeof...(u)) pr(" "); ps(u...); }
-
-// DEBUG
-void DBG() { cerr << "]" << endl; }
-tcTUU> void DBG(const T& t, const U&... u) {
-	cerr << ts(t); if (sizeof...(u)) cerr << ", ";
-	DBG(u...); }
-#ifdef LOCAL // compile with -DLOCAL, chk -> fake assert
-	#define dbg(...) cerr << "Line(" << __LINE__ << ") -> [" << #__VA_ARGS__ << "]: [", DBG(__VA_ARGS__)
-	#define chk(...) if (!(__VA_ARGS__)) cerr << "Line(" << __LINE__ << ") -> function(" \
-		 << __FUNCTION__  << ") -> CHK FAILED: (" << #__VA_ARGS__ << ")" << "\n", exit(0);
-#else
-	#define dbg(...) 0
-	#define chk(...) 0
-#endif
-
-void setPrec() { cout << fixed << setprecision(15); }
-void unsyncIO() { cin.tie(0)->sync_with_stdio(0); }
-// FILE I/O
-void setIn(str s) { freopen(s.c_str(),"r",stdin); }
-void setOut(str s) { freopen(s.c_str(),"w",stdout); }
-void setIO(str s = "") {
-	unsyncIO(); setPrec();
-	// cin.exceptions(cin.failbit); 
-	// throws exception when do smth illegal
-	// ex. try to read letter into int
-	if (sz(s)) setIn(s+".in"), setOut(s+".out"); // for USACO
-}
+//constant initialization
+const ll INF = 1e18; //Not too close to LLONG_MAX
 
+//variables used for the current problem
 int N;
 ll ans = 0;
-vpi v;
+vector<pi> v;
 
 void tri() {
 	sort(all(v));
-	vpi lef(N), rig(N);
-	auto comb = [](pi a, int b) -> pi { return {min(a.f,b),max(a.s,b)}; };
+	vector<pi> lef(N), rig(N);
+	auto comb = [](pi a, int b) -> pi { return mp(min(a.f,b),max(a.s,b)); };
 	lef[0] = {v[0].s,v[0].s};
-	FOR(i,1,N) lef[i] = comb(lef[i-1],v[i].s);
-	rig[N-1] = {v[N-1].s,v[N-1].s};
-	R0F(i,N-1) rig[i] = comb(rig[i+1],v[i].s);
-	ll area = (ll)(v.bk.f-v.ft.f)*(lef.bk.s-lef.bk.f);
-	ll bes = INF;
-	F0R(i,N-1) if (v[i].f != v[i+1].f)
-		ckmin(bes,(ll)(v[i].f-v[0].f)*(lef[i].s-lef[i].f)
-			+(ll)(v.bk.f-v[i+1].f)*(rig[i+1].s-rig[i+1].f));
-	ckmax(ans,area-bes);
+	for (int i = 1; i < N; ++i) lef[i] = comb(lef[i-1],v[i].s);
+	rig[N-1] = mp(v[N-1].s,v[N-1].s);
+	for (int i = N-2; i >= 0; --i) rig[i] = comb(rig[i+1],v[i].s);
+	ll area = (ll)(v.back().f-v.front().f)*(lef.back().s-lef.back().f);
+	ll best = INF;
+	for (int i = 0; i < N-1; ++i) if (v[i].f != v[i+1].f)
```

Comment:
Remove the ```if``` condition and everything will be fine

Reply:
when I remove this line, everything will be done.
@frodakcin

Reply:
> Remove the `if` condition and everything will be fine
Basically, that condition skips some of the cases which potentially are the answers.